File: project/app/src/main/java/org/owntracks/android/injection/modules/ServiceModule.java
package org.owntracks.android.injection.modules;


import android.app.Service;
import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.qualifier.ActivityContext;
import org.owntracks.android.injection.qualifier.AppContext;
import org.owntracks.android.injection.qualifier.ServiceContext;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.injection.scopes.PerApplication;
import org.owntracks.android.services.BackgroundService;

import dagger.Module;
import dagger.Provides;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ServiceModule {
    @Provides
    @PerActivity
    @ServiceContext
    static Service serviceContext(Service service) {
        return service;
    }

}

File: project/app/src/gms/java/org/owntracks/android/services/ServiceModule.kt
package org.owntracks.android.services

import android.app.Service
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent
import dagger.hilt.android.scopes.ServiceScoped
import org.owntracks.android.gms.location.GMSLocationProviderClient
import org.owntracks.android.gms.location.geofencing.GMSGeofencingClient
import org.owntracks.android.location.LocationProviderClient
import org.owntracks.android.location.geofencing.GeofencingClient

@InstallIn(ServiceComponent::class)
@Module
class ServiceModule {
    @Provides
    @ServiceScoped
    fun getGeofencingClient(service: Service): GeofencingClient =
        GMSGeofencingClient.create(service)

    @Provides
    @ServiceScoped
    fun getLocationProviderClient(service: Service): LocationProviderClient =
        GMSLocationProviderClient.create(service)

}